streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -73,22 +73,23 @@
     private final Set<String> globalNonPersistentStoresTopics = new HashSet<>();
     private final OffsetCheckpoint checkpointFile;
     private final Map<TopicPartition, Long> checkpointFileCache;
+    private final Map<String, String> storeToChangelogTopic;
 
     public GlobalStateManagerImpl(final LogContext logContext,
                                   final ProcessorTopology topology,
                                   final Consumer<byte[], byte[]> globalConsumer,
                                   final StateDirectory stateDirectory,
                                   final StateRestoreListener stateRestoreListener,
                                   final StreamsConfig config) {
+        storeToChangelogTopic = topology.storeToChangelogTopic();
```

Comment:
Wondering if we should pass `storeToChangelogTopic()` and `.globalStateStores()` into the constructor instead of `ProcessorTopology` ?

Reply:
I would still pass `ProcessorTopology` into the constructor because it might make the signature of the constructor more stable. I removed the field for the topology and now we store only `globalStateStores` and `storeChangelogTopics` in `GlobalStateManagerImpl`.